tree/ntuple/v7/inc/ROOT/RNTupleDescriptor.hxx

```diff
@@ -300,6 +303,59 @@ private:
    std::unordered_map<DescriptorId_t, RClusterDescriptor> fClusterDescriptors;
 
 public:
+   class RFieldDescriptorRange {
+   private:
+      const RNTupleDescriptor& fNTuple;
+      const RFieldDescriptor& fField;
+      std::vector<std::size_t> fOrdering = {};
+   public:
+      class RIterator {
+      private:
+         const RNTupleDescriptor& fNTuple;
+         const std::vector<DescriptorId_t>& fFieldChildren;
+         const std::vector<std::size_t>& fOrdering;
+         std::size_t fIndex = 0;
+      public:
+         using iterator_category = std::forward_iterator_tag;
+         using iterator = RIterator;
+         using value_type = RFieldDescriptor;
+         using difference_type = std::ptrdiff_t;
+         using pointer = RFieldDescriptor*;
+         using reference = const RFieldDescriptor&;
+
+         RIterator(const RNTupleDescriptor& ntuple, const RFieldDescriptor& fieldDesc,
+            const std::vector<std::size_t>& ordering, std::size_t index)
+            : fNTuple(ntuple), fFieldChildren(fieldDesc.GetLinkIds()), fOrdering(ordering), fIndex(index) {}
+         iterator operator++() { ++fIndex; return *this; }
+         reference operator*() {
+            return fNTuple.GetFieldDescriptor(
+               fFieldChildren.at(fOrdering.at(fIndex))
+            );
+         }
+         bool operator!=(const iterator& rh) const { return fIndex != rh.fIndex; }
+      };
+      /// An iterator over a field's children.
+      RFieldDescriptorRange(const RNTupleDescriptor& ntuple, const RFieldDescriptor& field)
+         : fNTuple(ntuple), fField(field), fOrdering(std::vector<std::size_t>(field.GetLinkIds().size()))
+      {
+         std::iota(fOrdering.begin(), fOrdering.end(), 0);
+      }
+      RFieldDescriptorRange SortByNames() {
+         RFieldDescriptorRange range(fNTuple, fField);
+         std::sort(range.fOrdering.begin(), range.fOrdering.end(), [this](auto left, auto right) {
```

Comment:
Not sure.. should this be `const auto &left` (and right)?

Reply:
The sort function is now taking `DescriptorId_t` so I don't think it makes a difference (and is nicer to type for people writing their own lambdas).